autoload/lsp/omni.vim

```diff
@@ -204,4 +219,63 @@ function! lsp#omni#get_vim_completion_item(item, ...) abort
     return l:completion
 endfunction
 
+augroup lsp_completion_item_text_edit
+    autocmd!
+    autocmd CompleteDone * call <SID>apply_text_edit()
+augroup END
+
+function! s:apply_text_edit() abort
+    " textEdit support function(callin from CompleteDone).
+    "
+    " expected user_data structure:
+    "     v:completed_item['user_data']: {
+    "       'type": 'vim-lsp/textEdit',
+    "       'content': {
+    "         'range': { ...(snip) },
+    "         'newText': 'yyy'
+    "        },
+    "     }
+    if !g:lsp_text_edit_enabled
+        return
+    endif
+
+    " completion faild or not select complete item
+    if empty(v:completed_item) || v:completed_item['word'] ==# ''
+        return
+    endif
+
+    " check user_data
+    if !has_key(v:completed_item, 'user_data') || empty(v:completed_item['user_data'])
+        return
+    endif
+
+    " check type
+    let l:user_data = eval(v:completed_item['user_data'])
+    if l:user_data['type'] !=# s:completion_type
```

Comment:
please add check existing `type` in `l:user_data`. user_data possible be returned empty.

Reply:
Thank you for pointing out. I fixed 421202982d1439ae593dcee9752ddd4bdbdb6669.